api-gateway/src/main/resources/static/scripts/app.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
+// Whitelist for all things related to auth and Q 401/403 handling
+const whiteList = new Set([
+    'login',
+    'signup',
+]);
+
 /* App Module */
 var petClinicApp = angular.module('petClinicApp', [
     'ui.router', 'layoutNav', 'layoutFooter', 'layoutWelcome', 'ownerList', 'ownerDetails', 'ownerForm', 'petForm'
     , 'visits', 'vetList','vetForm','vetDetails', 'loginForm', 'rolesDetails', 'signupForm', 'billDetails', 'billHistory'
     , 'verification']);
 
+petClinicApp.factory("authProvider", ["$window", function ($window) {
+
+    return {
+        setUser: ({ token, username, email }) => {
+            $window.localStorage.setItem("token", token)
+            $window.localStorage.setItem("username", username)
+            $window.localStorage.setItem("email", email)
+        },
+        getUser: () => ({
+            token: $window.localStorage.getItem("token"),
+            username: $window.localStorage.getItem("username"),
+            email: $window.localStorage.getItem("email"),
+        }),
+        purgeUser: () => {
+            $window.localStorage.removeItem("token")
+            $window.localStorage.removeItem("username")
+            $window.localStorage.removeItem("email")
+        },
+        isLoggedIn: () => !!$window.localStorage.getItem("token")
```

Comment:
Not sure I get what !! does here

Reply:
Im guessing it converts the value to a boolean? but how

Reply:
[Because of how truthy/falsy works](https://developer.mozilla.org/en-US/docs/Glossary/Truthy)